### PR TITLE
heartbeat endpoint

### PR DESCRIPTION
**Pull request by muhammad-ammar**
[EDUCATOR-1632](https://openedx.atlassian.net/browse/EDUCATOR-1632)

### CODE REVIEW

**Comment by muhammad-ammar** [comment]
@muzaffaryousaf @mushtaqak @Qubad786 This is ready for review.

File: VEDA_OS01/views.py [hunk 1]
```diff
@@ -284,3 +286,36 @@ def user_login(request):
...
+    """
+    View to check if database is reachable and ready to handle requests.
+    """
+    try:
+        db_status()
+    except DatabaseError as fail:
+        return JsonResponse(
+            {
+                'OK': False,
+                'check': 'db',
```
Comment (Qubad786) [inline VEDA_OS01/views.py]: reminder: only keep `{'OK': False}`

File: VEDA_OS01/tests/test_views.py [hunk 2]
```diff
@@ -221,3 +222,27 @@ def test_cielo24_error(self, mock_logger):
...
+    @patch('django.db.backends.utils.CursorWrapper')
+    def test_heartbeat_failure_db(self, mocked_cursor_wrapper):
+        """
+        Test that heartbeat endpoint gives expected response when there is an error.
+        """
+        error = 'Unable to connect to db'
+        mocked_cursor_wrapper.side_effect = DatabaseError(error)
+        response = self.client.get(reverse('heartbeat'))
+        assert response.status_code == 500
+        assert json.loads(response.content) == {'OK': False, 'check': 'db', 'error': error}
```
Comment (Qubad786) [inline VEDA_OS01/tests/test_views.py]: we can use `self.assertDict` here.

File: VEDA_OS01/tests/test_views.py [hunk 3]
```diff
@@ -221,3 +222,27 @@ def test_cielo24_error(self, mock_logger):
...
+    """
+    Tests for hearbeat endpoint.
+    """
+    def test_heartbeat(self):
+        """
+        Test that heartbeat endpoint gives expected response upon success.
+        """
+        response = self.client.get(reverse('heartbeat'))
+        assert response.status_code == 200
+        assert json.loads(response.content) == {'OK': True}
```
Comment (Qubad786) [inline VEDA_OS01/tests/test_views.py]: we can use `self.assertDict` here.

File: VEDA_OS01/views.py [hunk 4]
```diff
@@ -284,3 +286,36 @@ def user_login(request):
...
+
+@api_view(['GET'])
+@permission_classes([AllowAny])
+def heartbeat(request):  # pylint: disable=unused-argument
+    """
+    View to check if database is reachable and ready to handle requests.
+    """
+    try:
+        db_status()
+    except DatabaseError as fail:
```
Comment (mushtaqak) [inline VEDA_OS01/views.py]: I would prefer `error` or `exception`

File: VEDA_OS01/views.py (hunk 1 above)
Comment (mushtaqak) [inline VEDA_OS01/views.py]: Why are we sending `check: db` ?

File: VEDA_OS01/tests/test_views.py [hunk 5]
```diff
@@ -221,3 +222,27 @@ def test_cielo24_error(self, mock_logger):
...
+
+    @patch('django.db.backends.utils.CursorWrapper')
+    def test_heartbeat_failure_db(self, mocked_cursor_wrapper):
+        """
+        Test that heartbeat endpoint gives expected response when there is an error.
+        """
+        error = 'Unable to connect to db'
+        mocked_cursor_wrapper.side_effect = DatabaseError(error)
+        response = self.client.get(reverse('heartbeat'))
+        assert response.status_code == 500
```
Comment (mushtaqak) [inline VEDA_OS01/tests/test_views.py]: Nice :)

File: VEDA_OS01/tests/test_views.py (hunk 3 above)
Comment (muhammad-ammar) [reply VEDA_OS01/tests/test_views.py]: For pytests, we should use assert statement. @mushtaqak your thoughts, you were discussing something like this yesterday?

**Comment by muhammad-ammar** [comment]
@mushtaqak @Qubad786 Addressed your feedback

File: VEDA_OS01/tests/test_views.py (hunk 3 above)
Comment (Qubad786) [reply VEDA_OS01/tests/test_views.py]: @muhammad-ammar If we are already inheriting from `APITestCase` anyway, why shouldn't we use `self.assertXXX..`?

**Pull request merged**